builder/azure/arm/template_factory.go

```diff
@@ -51,6 +51,13 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 		builder.SetMarketPlaceImage(config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion)
 	}
 
+	if config.VirtualNetworkName != "" {
+		builder.SetVirtualNetwork(
+			config.VirtualNetworkName,
```

Comment:
I figured out why the unit tests did not catch this: poor choice of names.  All of the variables had values of "ignore," which meant the order was not really considered.  Fixed.